openmeetings-web/src/main/webapp/css/raw-tree.css

```diff
@@ -24,6 +23,90 @@
 .file-tree .footer .sizes .size {
 	padding-left: 5px;
 }
+/* Calc width with default for max level of 7 */
+.file-tree .file.item .name
+	, .file-tree .file.item .name span
+{
+	max-width: calc(var(--room-sidebar-width) - 26px);
+	min-width: 60px;
+}
+.file-tree .tree-subtree .file.item .name
```

Comment:
Am I right thinking these rules are for each level of nesting?
this is counter productive :(
there should be better solution ...

Reply:
I knew you would say that.

But if you want a proper fix which is probably rewriting the entire tree so that:
* the content resizes to the parent width by using DIV elements (not using span and making them block) 
* have the icon on the right of the span, using flex + row in each line item. Not absolute position
* use the ellipses for the text if not enough space cause by giving the span (inline) element no fixed width but putting it in between block elements. So that when it shrinks, it will shrink the span element as the only element that has no fixed width

And if the width is goes beyond a minimum width:
* change the tree to a plain list with some icons to save the space

That is probably how you would fix it. If you do it "proper"

So yes potentially my fix is technically its imperfect. I understand. You can't have endless number of tree-notes. 

But do you want to rewrite the entire tree?

Cause practically, if you go 7-levels down, there is no-way you will be able to see the file-name without resizing, cause the "cross" + indention + folder icons are already 7 times 18px = 126px. The user will not be able to see those files without making the sidebar bigger manually. And he probably will understand to drag the window until he can read it!

So I think given the circumstances and not rewrite the component basic structure it is probably acceptable. Unless you want to rewrite the actual DIV and SPAN structure.

Reply:
And honestly it is probably quite a lot better than what we have now which is toggling it between 230px width or 40px width.

Reply:
I would vote for implementing proper fix

I can try to do it tomorrow
I believe `display flex` is not necessary

Would you like me to try?

Reply:
@solomax can we commit above first? And refactor it in another PR?
It took me 2 days to work on that. You told me to use css calc. It wasn't my idea. I did it. And now you want me to throw it away again?

Reply:
There is no functional error with above. And it is an improvement. I'm sure there is also other  Jiras you can do. And we create another Jira for 5.0.0-M6 (or 5.0.1) for doing a more proper fix.

Reply:
@sebawagner I thought about this view a lot
you are right, this is most probably the only way to display this tree structure

maybe we should additionally

- add `min-width` to *each* tree item?
- add `min-width` to tree-panel with `overflow-x: auto`? (so in case there will be no space user can scroll)

WDYT?

Reply:
min-width: 60px; 
is already applied to every node. Its on the generic CSS rule that all tree nodes should have. So. I removed it and not repeat it into every subsequent rule for each tree level.

overflow-x: auto
I would need to rethink. Whats the point in calculating this size, if you scroll the entire panel anyway ? I thought the purpose is to have the warning icon visible. 
But yeah, probably you can still add a scrollbar horizontally, it won't be possible to avoid scrolling at some stage. And I think most users will expect to be able to scroll if its outside the visible area.

Reply:
Users are tend to ignore "exclamation mark" button, but it need to visible
I'm afraid it can be hidden if there are some levels of nesting and min-width will play

I guess it is partially visible on your "Resized to min" screen-shot

The only real concern having these CSS rules is: it will be not easy to maintain :(

Reply:
I make a new PR for subsequent changes
Thanks for your feedback!